x/group/module/abci_test.go

```diff
@@ -361,6 +359,159 @@ func TestEndBlocker(t *testing.T) {
+		GroupId: groupID,
+	}
+
+	err = policyReq.SetDecisionPolicy(policy)
+	require.NoError(t, err)
+	policyRes, err := app.GroupKeeper.CreateGroupPolicy(ctx, policyReq)
+	require.NoError(t, err)
+
+	groupPolicyAddr, err := sdk.AccAddressFromBech32(policyRes.Address)
+	require.NoError(t, err)
+
+	votingPeriod := policy.GetVotingPeriod()
+
+	msgSend := &banktypes.MsgSend{
+		FromAddress: groupPolicyAddr.String(),
+		ToAddress:   addrs[3].String(),
+		Amount:      sdk.Coins{sdk.NewInt64Coin("test", 100)},
+	}
+
+	proposers := []string{addrs[2].String()}
+
+	specs := map[string]struct {
+		preRun    func(sdkCtx sdk.Context) uint64
+		admin     string
+		expErrMsg string
+		newCtx    sdk.Context
+		expStatus group.ProposalStatus
+	}{
+		"proposal with status withdrawn is pruned after voting period end": {
+			preRun: func(sdkCtx sdk.Context) uint64 {
+				pId, err := submitProposal(app, sdkCtx, []sdk.Msg{msgSend}, proposers, groupPolicyAddr)
+				require.NoError(t, err)
+				_, err = app.GroupKeeper.WithdrawProposal(ctx, &group.MsgWithdrawProposal{
+					ProposalId: pId,
+					Address:    groupPolicyAddr.String(),
+				})
+				require.NoError(t, err)
+				return pId
+			},
+			admin:     proposers[0],
+			newCtx:    ctx.WithBlockTime(ctx.BlockTime().Add(votingPeriod).Add(time.Hour)),
+			expErrMsg: "load proposal: not found",
+			expStatus: group.PROPOSAL_STATUS_WITHDRAWN,
+		},
+		"proposal with status withdrawn is not pruned (before voting period)": {
+			preRun: func(sdkCtx sdk.Context) uint64 {
+				pId, err := submitProposal(app, sdkCtx, []sdk.Msg{msgSend}, proposers, groupPolicyAddr)
+				require.NoError(t, err)
+				_, err = app.GroupKeeper.WithdrawProposal(ctx, &group.MsgWithdrawProposal{
+					ProposalId: pId,
+					Address:    groupPolicyAddr.String(),
+				})
+				require.NoError(t, err)
+				return pId
+			},
+			admin:     proposers[0],
+			newCtx:    ctx,
+			expErrMsg: "",
+			expStatus: group.PROPOSAL_STATUS_WITHDRAWN,
+		},
+		"proposal with status aborted is pruned after voting period end (due to updated group policy decision policy)": {
+			preRun: func(sdkCtx sdk.Context) uint64 {
+				pId, err := submitProposal(app, sdkCtx, []sdk.Msg{msgSend}, proposers, groupPolicyAddr)
+				require.NoError(t, err)
+
+				policy := group.NewThresholdDecisionPolicy("3", time.Second, 0)
+				msg := &group.MsgUpdateGroupPolicyDecisionPolicy{
+					Admin:              addrs[0].String(),
+					GroupPolicyAddress: groupPolicyAddr.String(),
+				}
+				err = msg.SetDecisionPolicy(policy)
+				require.NoError(t, err)
+				_, err = app.GroupKeeper.UpdateGroupPolicyDecisionPolicy(ctx, msg)
+				require.NoError(t, err)
+
+				return pId
+			},
+			admin:     proposers[0],
+			newCtx:    ctx.WithBlockTime(ctx.BlockTime().Add(votingPeriod).Add(time.Hour)),
+			expErrMsg: "load proposal: not found",
+			expStatus: group.PROPOSAL_STATUS_ABORTED,
+		},
+		"proposal with status aborted is not pruned before voting period end (due to updated group policy)": {
```

Comment:
Do you think we can add these tests into the existing `TestEndBlocker`? Seems to me they are very similar

Reply:
They seem similar, but it's messing up things in `TestEndBlocker`
I think it's more cleaner this way